lib/hawkbit/hawkbit_device.c

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2020 Linumiz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "hawkbit_device.h"
+
+bool hawkbit_get_device_identity(char *id, int id_max_len)
+{
+	int i, id_len = 0;
+	char buf[3];
+	u8_t hwinfo_id[id_max_len];
+	size_t length;
+
+	length = hwinfo_get_device_id(hwinfo_id, sizeof(hwinfo_id) - 1);
+	if (length <= 0) {
+		return false;
+	}
+
+	memset(id, 0, id_max_len);
+
+	for (i = 0; i < length; i++) {
+		snprintk(buf, sizeof(buf), "%02x", hwinfo_id[i]);
+		id_len = strlen(id);
+		strncat(id, buf, id_max_len - id_len);
```

Comment:
Please check that you have enough space in id to copy the data

Reply:
Default size of id and hwinfo_id is 8. So, no need to check